getnacelle/nacelle-js
# chore: update API & types

## Why are these changes introduced?

Addresses [ENG-6848](https://nacelle.atlassian.net/browse/ENG-6848)

> Update Shopify API to `2022-07` & run `codegen`

## What is this pull request doing?

This PR updates all instances of the Shopify API from `2022-04` to `2022-07`.  All types are generated from the newer API version and mocks are updated.

## How to Test

1. Check out the `ENG-6848/update-shopify-api` branch.
2. `npm run bootstrap` from the monorepo root, as needed.
3. `cd packages/shopify-cart`
4. `npm run test:ci` - tests should pass with 100% coverage:

<img width="370" alt="Screen Shot 2022-08-26 at 10 09 26 AM" src="https://user-images.githubusercontent.com/24395931/186923861-9f202abd-caf8-4e48-808a-69c50459627b.png">

### E2E Checklist
Using the `thinx-test` vite project confirm the following mutations work.

- [x] `cartAttributesUpdate` 
<img width="494" alt="cartAttributesUpdate" src="https://user-images.githubusercontent.com/24395931/186939497-0e7fe536-1858-43ab-a10f-f10db084b0c2.png">

- [x] `cartBuyerIdentityUpdate`
<img width="455" alt="cartBuyerIdentityUpdate" src="https://user-images.githubusercontent.com/24395931/186939525-5a54b9ba-000a-436b-81a5-3ace9389adf8.png">

- [x] `cartCreate` 
<img width="463" alt="cartCreate" src="https://user-images.githubusercontent.com/24395931/186938723-c3e0a539-7441-4518-a73e-56fd48211f79.png">

- [x] `cartDiscountCodesUpdate`
<img width="451" alt="cartDiscountCodesUpdate" src="https://user-images.githubusercontent.com/24395931/186940417-d451f6b8-c740-40a4-a17e-522ebfe66f72.png">

- [x] `cartLinesAdd`
<img width="538" alt="cartLinesAdd" src="https://user-images.githubusercontent.com/24395931/186942168-d7b3b693-6b7f-4622-90da-984ce14f6c86.png">

- [x] `cartLinesRemove`
<img width="456" alt="cartLinesRemove" src="https://user-images.githubusercontent.com/24395931/186942187-a3334ecb-7606-4734-a919-8db94059fb6c.png">

- [x] `cartLinesUpdate`
<img width="522" alt="cartLinesUpdate" src="https://user-images.githubusercontent.com/24395931/186942214-fe1f8b08-23b4-4482-a3c6-9f36eff0f202.png">

- [x] `cartNoteUpdate`
<img width="451" alt="cartNoteUpdate" src="https://user-images.githubusercontent.com/24395931/186942249-f06618ee-b4b8-40b9-a9ba-d837b3f5dca5.png">
